webview-ui/src/ClusterProperties/AgentPoolDisplay.tsx

```diff
@@ -1,22 +1,54 @@
 import styles from "./ClusterProperties.module.css";
-import { AgentPoolProfileInfo } from "../../../src/webview-contract/webviewDefinitions/clusterProperties";
+import { AgentPoolProfileInfo, ClusterInfo } from "../../../src/webview-contract/webviewDefinitions/clusterProperties";
+import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";
+import { EventDef, vscode } from "./state";
+import { EventHandlers } from "../utilities/state";
 
 export interface AgentPoolDisplayProps {
-    profileInfo: AgentPoolProfileInfo
+    clusterInfo: ClusterInfo;
+    profileInfo: AgentPoolProfileInfo;
+    eventHandlers: EventHandlers<EventDef>;
+    clusterOperationRequested: boolean;
 }
 
+// Abort on anything except: Canceled, Failed, Succeeded
+// Check about deleting
+type ProvisionState = "CapturingSecurityVHDSnapshot" | "Creating" | "Deleting" | "RefreshingServicePrincipalProfile"
```

Comment:
This list is probably not exhaustive, and I'm afraid that readers may think it's supposed to be. We're only using a subset of these, which we _do_ hope to be exhaustive, i.e. the ones that indicate an operation is not in progress (I'm wanting to call these "terminal" states, although I'm not confident that's correct terminology).

Instead, we could declare a `const` collection of these terminal states, and use those instead, like:
```typescript
const terminalProvisioningStates = ["Canceled", "Failed", "Succeeded"] as const;
```

If we wanted to create a discriminated union type out of these (which isn't needed yet) we could use:
```typescript
type TerminalProvisioningState = typeof terminalProvisioningStates[number];
```